src/server/game/Movement/MovementGenerators/WaypointMovementGenerator.cpp

```diff
@@ -270,35 +270,71 @@ void WaypointMovementGenerator<Creature>::MovementInform(Creature* creature)
 uint32 FlightPathMovementGenerator::GetPathAtMapEnd() const
 {
     if (i_currentNode >= i_path.size())
+    {
         return i_path.size();
+    }
 
     uint32 curMapId = i_path[i_currentNode]->mapid;
     for (uint32 i = i_currentNode; i < i_path.size(); ++i)
+    {
         if (i_path[i]->mapid != curMapId)
+        {
             return i;
+        }
+    }
 
     return i_path.size();
 }
 
+#define SKIP_SPLINE_POINT_DISTANCE_SQ (40.0f * 40.0f)
+
+bool IsNodeIncludedInShortenedPath(TaxiPathNodeEntry const* p1, TaxiPathNodeEntry const* p2)
+{
+    return p1->mapid != p2->mapid || std::pow(p1->x - p2->x, 2) + std::pow(p1->y - p2->y, 2) > SKIP_SPLINE_POINT_DISTANCE_SQ;
+}
+
 void FlightPathMovementGenerator::LoadPath(Player* player)
 {
     _pointsForPathSwitch.clear();
-    std::vector<uint32> const& taxi = player->m_taxi.GetPath();
-    for (uint32 src = player->m_taxi.GetTaxiSegment(), dst = player->m_taxi.GetTaxiSegment() + 1; dst < taxi.size(); src = dst++)
+    std::deque<uint32> const& taxi = player->m_taxi.GetPath();
+    float discount = player->GetReputationPriceDiscount(player->m_taxi.GetFlightMasterFactionTemplate());
+    for (uint32 src = 0, dst = 1; dst < taxi.size(); src = dst++)
     {
         uint32 path, cost;
         sObjectMgr->GetTaxiPath(taxi[src], taxi[dst], path, cost);
         if (path > sTaxiPathNodesByPath.size())
+        {
             return;
+        }
 
         TaxiPathNodeList const& nodes = sTaxiPathNodesByPath[path];
         if (!nodes.empty())
         {
+            TaxiPathNodeEntry const* start = nodes[0];
+            TaxiPathNodeEntry const* end = nodes[nodes.size() - 1];
+            bool passedPreviousSegmentProximityCheck = false;
             for (uint32 i = 0; i < nodes.size(); ++i)
-                i_path.push_back(nodes[i]);
+            {
+                sMapMgr->CreateMap(nodes[i]->mapid, player)->SummonCreature(1, { nodes[i]->x, nodes[i]->y, nodes[i]->z, 0.0f })->SetLevel(i ? i : 1);
```

Comment:
Does this creature have a purpose outside debugging?

Reply:
I tried it, and the code can work after deleting it

Reply:
and it
``` player->TeleportTo(node->map_id, node->x, node->y, node->z, player->GetOrientation()); ```